_posts/2019/2019-11-03-github-actions-ci-for-rails-with-postgresql.md

```diff
@@ -0,0 +1,109 @@
+---
+title: "Use GitHub Actions for Rails CI with Postgres"
+description: "So many commits to get here"
+layout: article
+category: ruby
+image:
+  base: '2019/github-actions-ci-for-rails-with-postgresql'
+  alt: 'Green Octocat testing'
+
+---
+
+Writing tests alongside writing software is strongly recommended. It helps you protect against bugs, reduce the fear of scary deploys and, can even help you write better software.
+
+The Ruby and Rails community have always had a strong focus on testing. Typically that means having some sort of “continuous integration” system constantly running your test suite.
+
+There’s a bunch of good choices, but recently GitHub launched [Actions](https://github.com/features/actions) (in beta), which lets you run arbitrary workflows, including tests, after certain things happen to your git repository.
+
+It does need a little extra configuration to run your tests using PostgreSQL (my SQL database of choice), if you use `structure.sql` database dumps [you’ll need a slightly different configuration](/ruby/github-actions-ci-for-rails-with-postgresql-11-and-structure-sql).
+
+
+## Instead of...
+
+...using a separate CI service.
+
+
+## Use...
+
+...Github Actions to run your Rails tests.
+
+### `.github/workflows/tests.yml`
+
+```yml
+name: Rails Tests
+
+on:
+  pull_request:
+    - 'master'
+  push:
+    - 'master'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    services:
+      postgres:
+        image: postgres:11.5
+        ports: ["5432:5432"]
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+
+    steps:
+    - uses: actions/checkout@v1
+
+    - name: Set up Ruby 2.6
+      uses: actions/setup-ruby@v1
+      with:
+        ruby-version: 2.6.x
+
+    - name: Install PostgreSQL 11 client
+      run: |
+        sudo apt-get -yqq install libpq-dev
+
+    - name: Build App
+      env:
+        PGHOST: localhost
+        PGUSER: postgres
+        RAILS_ENV: test
+      run: |
+        gem install bundler
+        bundle install --jobs 4 --retry 3
+        bin/rails db:setup
+
+    - name: Run Tests
+      env:
+        PGHOST: localhost
+        PGUSER: postgres
+        RAILS_ENV: test
+      run: |
+      bundle exec rake test
+      bundle exec rake test:system
+      # or Rspec
+      # bundle exec rspec
+```
+
+If you use SQL dumps—you have a `db/structure.sql` file rather than a `db/schema.rb` file in your application—you’ll need to use [this slightly altered workflow](github-actions-ci-for-rails-with-postgresql-11-and-structure-sql).
+
+
+## Why?
+
+There is a benefit to your tests being in the ‘same place’ as your code. There is no separate account to maintain, no separate service to bill, and no separate ‘team member management’.
+
+It's completely free for public, open source, repositories. There’s also a pretty generous free usage tier for private repositories, and pretty economical after that.
```

Comment:
*GitHub Actions is completely free... (since we haven't said the name in a while)

Reply:
*free-usage tier or just free tier?

Reply:
*, and it's pretty economical beyond that.